worker/jobs/bulkUpsertSamplesJob.js

```diff
@@ -76,10 +69,10 @@ module.exports = (job, done) => {
         const jobEndTime = Date.now();
 
         // number of successful upserts
-        objToReturn.recordCount = upsertResults.length - errorCount;
+        objToReturn.recordCount = successCount - errors.length;
```

Comment:
I don't think the failing tests were flappers - they weren't failing before this change
You added a successCount variable that is only incremented for a succesful result, but the recordCount is still being calculated by subtracting errors.length. 
It should just be recordCount = errors.length.